@xen-orchestra/xapi/src/vm.js

```diff
@@ -402,6 +404,26 @@ module.exports = class Vm {
     }
   }
 
+  @defer
+  // this method ignores VDIs which their names start with '[NOBAK]'
+  async customSnapshot($defer, vmRef, { cancelToken = CancelToken.none, nameLabel } = {}) {
+    assert.strictEqual(await this.getField('VM', vmRef, 'power_state'), 'Halted')
+
+    await asyncMap(this.getField('VM', vmRef, 'VBDs'), async vbdRef => {
+      const vbd = this.getObject(vbdRef)
+      if (
+        vbd.type === 'Disk' &&
+        isValidRef(vbd.VDI) &&
+        (await this.getField('VDI', vbd.VDI, 'name_label')).startsWith('[NOBAK]')
+      ) {
+        await this.VBD_destroy(vbdRef)
+        $defer.call(this, 'VBD_create', vbd)
+      }
+    })
+
+    return this.VM_snapshot(cancelToken, vmRef, nameLabel)
+  }
```

Comment:
Question: Do you think we could integrate this in the regular snapshot method if we detect the VM is halted and some disks contain the `[NOBAK]` flag?

Maybe it should be disablable via a library option.

Reply:
Why not.

Reply:
How you want to do it?